frontend/src/hooks/useGetAptogotchiCollection.ts

```diff
@@ -0,0 +1,88 @@
+import { useCallback, useState } from "react";
+import { useWallet } from "@aptos-labs/wallet-adapter-react";
+import { getAptosClient } from "@/utils/aptosClient";
+import { NEXT_PUBLIC_CONTRACT_ADDRESS } from "@/utils/env";
+import { queryAptogotchiCollection } from "@/graphql/queryAptogotchiCollection";
+import { padAddressIfNeeded } from "@/utils/address";
+
+const aptosClient = getAptosClient();
+
+type Collection = {
+  collection_id: string;
+  collection_name: string;
+  creator_address: string;
+  uri: string;
+  current_supply: any;
+};
+
+type CollectionHolder = {
+  owner_address: string;
+};
+
+type CollectionResponse = {
+  current_collections_v2: Collection[];
+  current_collection_ownership_v2_view: CollectionHolder[];
+};
+
+export function useGetAptogotchiCollection() {
+  const { account } = useWallet();
+  const [collection, setCollection] = useState<Collection>();
+  const [firstFewAptogotchiName, setFirstFewAptogotchiName] =
+    useState<string[]>();
+  const [loading, setLoading] = useState(false);
+
+  const fetchCollection = useCallback(async () => {
+    if (!account?.address) return;
+
+    try {
+      setLoading(true);
+
+      const aptogotchiCollectionIDResponse = (await aptosClient.view({
+        payload: {
+          function: `${NEXT_PUBLIC_CONTRACT_ADDRESS}::main::get_aptogotchi_collection_id`,
+          arguments: [],
+        },
+      })) as [`0x${string}`];
+
+      const collectionIDAddr = padAddressIfNeeded(
+        aptogotchiCollectionIDResponse[0]
+      );
+
+      console.log("collectionIDAddr: ", collectionIDAddr);
+
+      const collectionResponse: CollectionResponse =
+        await aptosClient.queryIndexer({
+          query: {
+            query: queryAptogotchiCollection,
+            variables: {
+              collection_id: collectionIDAddr,
+            },
+          },
+        });
+
+      const firstFewAptogotchi = await Promise.all(
+        collectionResponse.current_collection_ownership_v2_view
+          .filter((holder) => holder.owner_address !== account.address)
+          // TODO: change to limit 3 in gql after indexer fix limit
+          .slice(0, 3)
```

Comment:
nit: do we want to make 3 a constant?

Reply:
Haha i had a constant in an earlier version of the code and somehow changed it back when working on something else by accident. I'll make the change in the PR to address the TODO.